qa/suites/orch/cephadm/workunits/task/test_set_mon_crush_locations.yaml

```diff
@@ -0,0 +1,39 @@
+roles:
+- - host.a
+  - osd.0
+  - mon.a
+  - mgr.a
+- - host.b
+  - osd.1
+  - mon.b
+  - mgr.b
+- - host.c
+  - osd.2
+  - mon.c
+tasks:
+- install:
+- cephadm:
+- cephadm.apply:
+    specs:
+      - service_type: mon
+        service_id: foo
+        placement:
+          count: 3
+        spec:
+          crush_locations:
+            host.a:
+              - datacenter=a
+            host.b:
+              - datacenter=b
+              - rack=2
+            host.c:
+              - datacenter=a
+              - rack=3
+- cephadm.shell:
+    host.a:
+      - ceph orch ps --refresh
+      - sleep 180
+      - ceph mon dump --format json
+      - ceph mon dump --format json | jq -e '.mons | .[] | select(.name == "host.a") | .crush_location == "{datacenter=a}"'
```

Comment:
```
teuthology.exceptions.CommandFailedError: Command failed on smithi080 with status 4: 'sudo /home/ubuntu/cephtest/cephadm --image quay-quay-quay.apps.os.sepia.ceph.com/ceph-ci/ceph:749c36e80e1d077a46b710d0776d199145af6283 shell -c /etc/ceph/ceph.conf -k /etc/ceph/ceph.client.admin.keyring --fsid 8938d180-9ada-11ed-9e55-001a4aab830c -- bash -c \'ceph mon dump --format json | jq -e \'"\'"\'.mons | .[] | select(.name == "host.a") | .crush_location == "{datacenter=a}"\'"\'"\'\''
```
from right before,  `ceph mon dump --format json` output
```
{"epoch":3,"fsid":"8938d180-9ada-11ed-9e55-001a4aab830c","modified":"2023-01-23T05:00:41.427412Z","created":"2023-01-23T04:59:00.154063Z","min_mon_release":18,"min_mon_release_name":"reef","election_strategy":1,"disallowed_leaders: ":"","stretch_mode":false,"tiebreaker_mon":"","removed_ranks: ":"","features":{"persistent":["kraken","luminous","mimic","osdmap-prune","nautilus","octopus","pacific","elector-pinging","quincy","reef"],"optional":[]},"mons":[{"rank":0,"name":"a","public_addrs":{"addrvec":[{"type":"v2","addr":"172.21.15.80:3300","nonce":0},{"type":"v1","addr":"172.21.15.80:6789","nonce":0}]},"addr":"172.21.15.80:6789/0","public_addr":"172.21.15.80:6789/0","priority":0,"weight":0,"crush_location":"{}"},{"rank":1,"name":"c","public_addrs":{"addrvec":[{"type":"v2","addr":"172.21.15.195:3300","nonce":0},{"type":"v1","addr":"172.21.15.195:6789","nonce":0}]},"addr":"172.21.15.195:6789/0","public_addr":"172.21.15.195:6789/0","priority":0,"weight":0,"crush_location":"{}"},{"rank":2,"name":"b","public_addrs":{"addrvec":[{"type":"v2","addr":"172.21.15.87:3300","nonce":0}
```
so crush locations were not set it seems

Reply:
I think this is because the spec is using hosts like `host.a` but cephadm knows the hosts by names like `smithi001` so it doesn't think it's deploying the mons on the hosts the spec mentions